recipes/boost/all/conanfile.py

```diff
@@ -141,25 +208,73 @@ def configure(self):
                 if not self.options.get_safe('without_%s' % lib):
                     raise ConanInvalidConfiguration("Boost '%s' library requires multi threading" % lib)
 
-        if self.settings.compiler == "Visual Studio" and "MT" in self.settings.compiler.runtime and self.options.shared:
+        if self.settings.compiler == "Visual Studio" and "MT" in str(self.settings.compiler.runtime) and self.options.shared:
             raise ConanInvalidConfiguration("Boost can not be built as shared library with MT runtime.")
 
+        # Check, when a boost module is enabled, whether the boost modules it depends on are enabled as well.
+        for mod_name, mod_deps in self._dependencies["dependencies"].items():
+            if not self.options.get_safe("without_{}".format(mod_name), True):
+                for mod_dep in mod_deps:
+                    if self.options.get_safe("without_{}".format(mod_dep), False):
+                        raise ConanInvalidConfiguration("{} requires {}: {} is disabled".format(mod_name, mod_deps, mod_dep))
```

Comment:
Due to the number of options, an improvement could be to accumulate missing dependencies and display a message showing a summary of missing components.

Reply:
During build a list of all the components with `building` or `not building` next to each one is shown.

Reply:
Is that ok for you @SpaceIm ?

Reply:
Yes I guess, let's eventually improve it later.